src/main/java/org/opensearch/neuralsearch/rest/RestNeuralStatsHandler.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.rest;
+
+import com.google.common.collect.ImmutableList;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang.StringUtils;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.neuralsearch.settings.NeuralSearchSettingsAccessor;
+import org.opensearch.neuralsearch.stats.NeuralStatsInput;
+import org.opensearch.neuralsearch.stats.events.EventStatName;
+import org.opensearch.neuralsearch.stats.state.StateStatName;
+import org.opensearch.neuralsearch.transport.NeuralStatsAction;
+import org.opensearch.neuralsearch.transport.NeuralStatsRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestActions;
+import org.opensearch.transport.client.node.NodeClient;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.neuralsearch.plugin.NeuralSearch.NEURAL_BASE_URI;
+import static org.opensearch.neuralsearch.processor.util.RestActionUtils.splitCommaSeparatedParam;
+
+@Log4j2
+@AllArgsConstructor
+public class RestNeuralStatsHandler extends BaseRestHandler {
+    private static final String NAME = "neural_stats_action";
+    public static final String FLATTEN_PARAM = "flat_keys";
+    public static final String INCLUDE_METADATA_PARAM = "include_metadata";
+
+    private static final Set<String> EVENT_STAT_NAMES = EnumSet.allOf(EventStatName.class)
+        .stream()
+        .map(EventStatName::getNameString)
+        .map(String::toLowerCase)
+        .collect(Collectors.toSet());
+
+    private static final Set<String> STATE_STAT_NAMES = EnumSet.allOf(StateStatName.class)
```

Comment:
Can you come up with more meaninful name? State_stat is not that reader friendly.

Reply:
Sure, I'll rename to `InfoStat`